specification/azurearcdata/resource-manager/Microsoft.AzureArcData/preview/2023-01-15-preview/failoverGroups.json

```diff
@@ -0,0 +1,231 @@
+        "provisioningState": {
+          "type": "string",
+          "enum": [
+            "Succeeded",
+            "Failed",
+            "Canceled",
+            "Accepted"
+          ],
+          "x-ms-enum": {
+            "name": "provisioningState",
+            "modelAsString": true
+          },
+          "description": "The provisioning state of the failover group resource.",
+          "readOnly": true
+        },
+        "partnerManagedInstanceId": {
+          "type": "string",
+          "description": "The resource ID of the partner SQL managed instance."
+        },
+        "spec": {
+          "$ref": "#/definitions/FailoverGroupSpec",
+          "description": "null",
+          "x-ms-client-flatten": false
+        },
+        "status": {
+          "$ref": "#/definitions/FailoverGroupStatus",
+          "description": "null",
+          "x-ms-client-flatten": false
+        }
+      },
+      "required": [
+        "partnerManagedInstanceId",
+        "spec"
+      ]
+    },
+    "FailoverGroupSpec": {
+      "description": "The specifications of the failover group resource.",
+      "type": "object",
+      "properties": {
+        "sharedName": {
+          "type": "string",
+          "description": "The shared name of the failover group for this SQL managed instance. Both SQL managed instance and its partner have to use the same shared name."
+        },
+        "sourceMI": {
+          "type": "string",
+          "description": "The name of the SQL managed instance with this failover group role."
+        },
+        "partnerMI": {
+          "type": "string",
+          "description": "The name of the partner SQL managed instance."
+        },
+        "partnerMirroringURL": {
+          "type": "string",
+          "description": "The mirroring endpoint URL of the partner SQL managed instance."
+        },
+        "partnerMirroringCert": {
+          "type": "string",
+          "description": "The mirroring endpoint public certificate for the partner SQL managed instance. Only PEM format is supported."
+        },
+        "partnerSyncMode": {
+          "type": "string",
+          "enum": [
+            "async",
+            "sync"
+          ],
+          "x-ms-enum": {
+            "name": "failoverGroupPartnerSyncMode",
+            "modelAsString": true
+          },
+          "default": "async",
+          "description": "The partner sync mode of the SQL managed instance."
+        },
+        "role": {
+          "type": "string",
+          "enum": [
+            "primary",
+            "secondary",
+            "force-primary-allow-data-loss",
+            "force-secondary"
+          ],
+          "x-ms-enum": {
+            "name": "instanceFailoverGroupRole",
+            "modelAsString": true
+          },
+          "default": "primary",
+          "description": "The role of the SQL managed instance in this failover group."
+        }
+      },
+      "additionalProperties": {},
```

Comment:
>additionalProperties [](http://example.com/codeflow?start=7&length=20)

What is this meant for?

Reply:
There may be spec fields set by the customer on the cluster side that aren't defined here under properties, so we were thinking we could add this here so that when the resource is synced to azure, the customer-added properties will be surfaced.